Task 27 forgotten password

OlegPanasyuk opened the pull request.

OlegPanasyuk committed:
Add Form Forgot Passowd

OlegPanasyuk committed:
Forgot passord and reset

OlegPanasyuk committed:
Merge branch 'dev' into Task-27-Forrgot

OlegPanasyuk committed:
Fix  some comments from conversation

OlegPanasyuk committed:
Merge

OlegPanasyuk committed:
Add validation to forms

slavasgod commented on src/components/spinner/index.js:
```diff
@@ -0,0 +1,14 @@
+        lds_roller
+    } = styles;
+    return (
+        <div className={lds_roller}><div></div><div></div><div></div><div></div><div></div><div></div><div></div><div></div></div>
```
What is it?

OlegPanasyuk replied:
It's a CTRL+C CTRL+V

slavasgod commented on src/components/spinner/style.module.scss:
```diff
@@ -0,0 +1,104 @@
+  height: 64px;
+}
+
+.lds_roller div {
```
Use scss syntax pls

slavasgod commented on src/containers/auth/forgotPassword/index.jsx:
```diff
@@ -0,0 +1,162 @@
+
+  refInput = React.createRef();
+
+  send = async email => {
```
Why this function is async?

OlegPanasyuk replied:
Because of, inner functions are async and I need to wait for the response from the Server.

slavasgod commented on src/containers/auth/forgotPassword/index.jsx:
```diff
@@ -0,0 +1,162 @@
+            <button
+              type="submit"
+              className={btn}
+              onClick={e => {
```
Remove function implementation from jsx

OlegPanasyuk replied:
Ok

slavasgod commented on src/containers/auth/forgotPassword/styles.module.scss:
```diff
@@ -0,0 +1,80 @@
+
+.modal {
+    font-family: $primary-font;
+    background-color: $bcg-gray-dark;
```
Please, use stylelint

OlegPanasyuk replied:
It doesn't work

OlegPanasyuk replied:
But now it works...

slavasgod commented on src/containers/auth/forgotPassword/styles.module.scss:
```diff
@@ -0,0 +1,80 @@
+        margin: 15px 0 0 0;
+    }
+
+    .rowTop {
```
We have another codestyle rules in css: row_top

OlegPanasyuk replied:
done

slavasgod commented on src/containers/auth/resetPassword/index.jsx:
```diff
@@ -0,0 +1,137 @@
+        passwordConfirm: null,
+        linkId: null,
+        message: null
+    }
```
Use prettier

OlegPanasyuk replied:
done

slavasgod commented on src/containers/auth/resetPassword/index.jsx:
```diff
@@ -0,0 +1,137 @@
+        }
+    }
+
+    sendReset = async (e) => {
```
Remove async

OlegPanasyuk replied:
Without async, it doesn't work. Because of Axios return a promise and it needs to wait to resolve one.

slavasgod commented on src/containers/auth/resetPassword/index.jsx:
```diff
@@ -0,0 +1,137 @@
+                    });
+                } else {
+                    const { status, data } = await resetPassword({ linkId, password, passwordConfirm });
+                    if (data.status === 200) {
```
You should rewrite this 4 `if`. It looks awful

OlegPanasyuk replied:
Yep. I agree with you.

OlegPanasyuk replied:
rewrited

slavasgod commented on src/containers/auth/resetPassword/styles.module.scss:
```diff
@@ -0,0 +1,66 @@
+    margin: 15px 0 0 0;
+  }
+
+  .rowTop {
```
Codestyle in css...

OlegPanasyuk replied:
done

slavasgod commented on src/services/auth.js:
```diff
@@ -35,3 +35,48 @@ export const googleLoginRequest = async data => {
+
+export const resetApprove = async linkId => {
+  try {
+    const l = encodeURIComponent(linkId);
```
Pls, use sensible variables

OlegPanasyuk replied:
Ok

slavasgod commented on stories/index.js:
```diff
@@ -0,0 +1,18 @@
+/* eslint-disable import/no-extraneous-dependencies */
```
Why are you disable eslint rules?

slavasgod commented on .eslintrc:
```diff
@@ -22,7 +22,8 @@
     "import/imports-first": ["error", "absolute-first"],
-    "import/newline-after-import": "error"
+    "import/newline-after-import": "error",
+    "react/state-in-constructor": ["warn", "never"]
```
Why you add this rule? It's not bad to use constructors.

OlegPanasyuk replied:
Because eslint underline my code.

slavasgod commented on src/components/spinner/index.js:
```diff
@@ -0,0 +1,14 @@
+import styles from './style.module.scss';
+
+
+const Spinner = () => {
```
I think we should use libraries for spinner. This spinner atleast not flexible

slavasgod commented on src/containers/auth/forgotPassword/index.jsx:
```diff
@@ -0,0 +1,162 @@
+                onChange={this.onChangeInput}
+                onInput={() =>
+                  this.setState({
+                    message: null,
```
Remove function implementation from jsx

OlegPanasyuk replied:
Ok

slavasgod commented on src/containers/auth/forgotPassword/styles.module.scss:
```diff
@@ -0,0 +1,80 @@
+    }
+
+    .closeModal {
+        color: white;
```
Magic color?

slavasgod commented on src/components/spinner/index.js:
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import styles from './style.module.scss';
```
Enter a whitespace between this imports and replace imports order in other files like: main react libraries - custom react libraries - your components - styles

slavasgod commented on src/containers/auth/forgotPassword/index.jsx:
```diff
@@ -0,0 +1,162 @@
+      errors.email &&
+      this.setState({
+        message: errors.email,
+      });
```
How this answer variable works? You declare it, give it value and then replace this value?

OlegPanasyuk replied:
It maid to break notifications from Eslint

slavasgod commented on src/containers/auth/resetPassword/index.jsx:
```diff
@@ -0,0 +1,137 @@
+                        </div>
+
+                    </form>)}
+                {(status) && (!success) && (status > 300) && (<div className={`${big_size_message} ${errorMessage}`}>Something is wrong</div>)}
```
Make this logic easier pls. It is hard to understand

OlegPanasyuk replied:
done

OlegPanasyuk committed:
Refactor code. Change Spinner.

OlegPanasyuk committed:
Some code fixies

slavasgod commented on package.json:
```diff
@@ -40,20 +41,27 @@
-    "redux-saga": "^1.0.5"
+    "redux-saga": "^1.0.5",
+    "styled-components": "^4.3.2"
   }
```
1. We should talk about using bootstrap.
2. Why you use 2 different spinners libraries?
3. I have no idea why you add jquery and styled-components to packages.

OlegPanasyuk replied:
done

slavasgod commented on src/containers/auth/forgotPassword/index.jsx:
```diff
@@ -0,0 +1,165 @@
+            </ShowMessage>
+          </ShowMessage>
+          <ShowMessage
+            classStyle={`${row} ${rowFlexEnd}`}
```
rowFlexEnd - row_flex_end

OlegPanasyuk replied:
No, we have a rule in eslint and we need to use it.

slavasgod commented on src/containers/auth/forgotPassword/styles.module.scss:
```diff
@@ -0,0 +1,85 @@
+    color: $pink;
+  }
+
+  input:invalid {
```
Use scss syntax:
```
input {
&:invalid { }
}
```

Also use scss nesting

OlegPanasyuk replied:
Ok

slavasgod commented on src/containers/auth/login/index.js:
```diff
@@ -95,17 +96,25 @@ class Login extends Component {
+            <span 
+              onClick={() => {
+                this.closeModal();
+                onModalCloseForgotPass(true);
```
Don't implement functions in jsx!

OlegPanasyuk replied:
Ok

slavasgod replied:
!

OlegPanasyuk replied:
done

slavasgod commented on src/containers/auth/resetPassword/index.jsx:
```diff
@@ -0,0 +1,143 @@
+    } = styles;
+
+    return (
+      <div className={wrapperF}>
```
Use sensible names of classes, what is wrapperF?

OlegPanasyuk replied:
Ok

OlegPanasyuk replied:
done

slavasgod commented on src/containers/auth/resetPassword/index.jsx:
```diff
@@ -0,0 +1,143 @@
+              />
+            </div>
+
+            <ShowMessage classStyle={`${errorMessage}`} condition={!!message}>
```
If I understood, message is a string, so why you try to make boolean from it?

OlegPanasyuk replied:
Because I need to be sure that the condition prop is a Boolean.

OlegPanasyuk replied:
Rebuild. Breake this double !!

slavasgod commented on src/containers/auth/resetPassword/index.jsx:
```diff
@@ -0,0 +1,143 @@
+        <ShowMessage
+          classStyle={`${bigSizeMessage} ${errorMessage}`}
+          condition={!status && !success}>
+          <CircleSpinner color="#fff" size={40} loading={!status && !success} />
```
First of all, spinner must be migrated to special component, so we can use it at different places. I think you should mock component for spinner and use it if you want, because we didn't decide what spinner we will use.

OlegPanasyuk replied:
Ok, you are right. When we will have this spinner we will implement this. And it will be the necessary task

slavasgod commented on src/containers/auth/resetPassword/styles.module.scss:
```diff
@@ -0,0 +1,53 @@
+@import '../../../scss/combineStyles.scss';
+
+.wrapper_f {
```
Use sensible variables

OlegPanasyuk replied:
Ok

slavasgod commented on src/containers/auth/forgotPassword/index.jsx:
```diff
@@ -0,0 +1,165 @@
+  onSubmit = e => {
+    e.preventDefault();
+    const { email } = this.state;
+    this.send(email);
```
Why you always try to destruct object? One variable used only in one place, just use `this.state.email`

OlegPanasyuk replied:
Because it makes a reading of code easier.

slavasgod replied:
You are right, if there were a lot of variables or variable, that will used a lot of times. But in this case I don't agree

OlegPanasyuk committed:
Fix Slava's reviews

slavasgod commented on src/containers/auth/forgotPassword/index.jsx:
```diff
@@ -0,0 +1,165 @@
+  modalClose = () => {
+    const { onModalClose } = this.props;
+    this.onCloseModalSetDefault();
+    onModalClose(false);
```
Same

slavasgod commented on src/containers/auth/login/index.js:
```diff
@@ -95,17 +96,25 @@ class Login extends Component {
+                this.closeModal();
+                onModalCloseForgotPass(true);
+              }}
+              className={styles.linkForgot}
```
!

slavasgod commented on src/containers/auth/styles.module.scss:
```diff
@@ -64,6 +66,23 @@
     outline: inherit;
   }
+
+  .linkForgot {
```
!

OlegPanasyuk replied:
done

slavasgod commented on src/services/auth.js:
```diff
@@ -35,3 +35,48 @@ export const googleLoginRequest = async data => {
+    return {
+      status: 400,
+      data: 'Link is uncorrected',
+    };
```
I think you should make classes on backend side to throw errors, and catch them, not like returning default hardcoded objects

OlegPanasyuk replied:
Because of axios. If error middleware throw error. Axios throw error too. And don't return the right response with an error.

slavasgod replied:
Are you sure? I think you should try again.

slavasgod replied:
Make normal answer on backend side

slavasgod commented on src/containers/auth/forgotPassword/index.jsx:
```diff
@@ -0,0 +1,165 @@
+    const email = e.target.value;
+    this.setState({
+      email,
+    });
```
You can simplify this code like:
```
 onChangeInput = e => {
    this.setState({
      email: e.target.value
    });
```
or make this code more flexible, like:
```
 onChangeInput = e => {
    this.setState({
      [e.target.name]: e.target.value
    });
```

OlegPanasyuk replied:
done

slavasgod commented on src/containers/auth/forgotPassword/styles.module.scss:
```diff
@@ -0,0 +1,86 @@
+    cursor: pointer;
+    outline: inherit;
+  }
+}
```
Why you not use scss nesting?

OlegPanasyuk committed:
Fix the latest mistakes

OlegPanasyuk committed:
Adding Spinner wrapper

-	Adding component Spinner

-	Fixing calculating in props. Move it into variables

-	Adding PropTypes

slavasgod commented on src/containers/auth/forgotPassword/index.jsx:
```diff
@@ -0,0 +1,168 @@
+import { forgotPasswordRequest } from '../../../services/auth';
+import validateAuth from '../../../validation/auth';
+import ShowMessage from '../showMessage';
+import Spinner from '../../../components/spinner';
```
Fix import order: main react libraries- custom react libraries/default libraries - your components - styles. Separate each group by whitespace. Check all your files.

slavasgod commented on src/containers/auth/forgotPassword/index.jsx:
```diff
@@ -0,0 +1,168 @@
+
+    let showValidationMessage = !!message;
+    let showForm = !status && !pending;
+    let showSpinner = pending;
```
I think we decided to remove this boolean transformation. Just use message as a string.

slavasgod commented on src/containers/auth/forgotPassword/index.jsx:
```diff
@@ -0,0 +1,168 @@
+    let showSpinner = pending;
+    let showErrorMessage = (status >= 300);
+    let showSuccessMessage = (status < 300);
+    let showCheckMark = !status;
```
I don't like how you implement logic here. A lot of variables, all logic connected with statuses, that kind a strange. Maybe you should try to do it better. 
Also use const instead of let

slavasgod commented on src/containers/auth/resetPassword/index.jsx:
```diff
@@ -0,0 +1,148 @@
+  sendReset = async (e) => {
+    e.preventDefault();
+    const { linkId, password, passwordConfirm } = this.state;
+    let stateNew = { status: null, success: null };
```
Use const instead of let. Maybe it will be better `newState` ?

slavasgod commented on src/containers/auth/resetPassword/index.jsx:
```diff
@@ -0,0 +1,148 @@
+    let showErrorNotification = !!message;
+    let showLinkIncorrect = !!status && !success && (status > 300);
+    let showSpinner = !status && !success;
+    let showSuccessMessage = !!success;
```
Same as in file forgotPassword/index.jsx.

OlegPanasyuk committed:
Fix regex validation and link

OlegPanasyuk committed:
Merge dev into Task-27

OlegPanasyuk committed:
Rewriting components

-	Adding stateless Input, Button, Form components

-	Rewrite FrogotPassword and ResetPassword with pushing logic from render function to methods of class.
	Rewrite state of those components.

-	Adding into the service processing of axios reqests. validateStatus

OlegPanasyuk committed:
Megre dev to Task-27 branch

OlegPanasyuk committed:
Fixing UI problems with forgot password

OlegPanasyuk committed:
Adding onChange handler for inputs

melonholy commented on src/components/button/index.js:
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+
+export default function index({ type, className, onClick, children }) {
```
What does this function mean and why you are not using arrow functions?

OlegPanasyuk replied:
this is a react snippets) I didn't make attention on this case( I think i need to convert it to an arrow function

melonholy commented on src/components/form/index.js:
```diff
@@ -0,0 +1,5 @@
+import React from 'react';
+
+export default function index({ className, children }) {
```
Check comment above

melonholy commented on src/components/input/index.js:
```diff
@@ -0,0 +1,5 @@
+import React from 'react';
+
+export default function index(props) {
```
And once again)

melonholy commented on src/components/main/index.js:
```diff
@@ -6,19 +6,23 @@ import { GettingStarted } from '../help/gettingStarted';
-      <Route path="/troubleshooting" component={Trobleshooting} />
-      <Route path="/course-taking" component={CourseTaking} />
-    </Switch>
+    <>
```
You don't need Fragment here.Switch component performs this function

OlegPanasyuk replied:
Done

melonholy commented on src/containers/auth/forgotPassword/index.js:
```diff
@@ -0,0 +1,171 @@
+    },
+  };
+
+  showCurrentMessage = (keys = [], message, pending = false) => {
```
why you assign empty array?

OlegPanasyuk replied:
Done

melonholy commented on src/containers/auth/forgotPassword/index.js:
```diff
@@ -0,0 +1,171 @@
+
+  preValidateForm = ({ email }) => {
+    const errors = validateAuth({ email });
+    if (Object.keys(errors).length > 0) {
```
You don't need additional operation here, just check object for emptiness

OlegPanasyuk replied:
Done

melonholy commented on src/containers/auth/forgotPassword/styles.module.scss:
```diff
@@ -0,0 +1,98 @@
+  .row {
+    display: flex;
+    flex-wrap: wrap;
+    margin: 15px 0 0 0;
```
just use `margin-top`

melonholy commented on src/containers/auth/forgotPassword/styles.module.scss:
```diff
@@ -0,0 +1,98 @@
+  }
+
+  .row_top {
+    margin: 25px 0 0 0;
```
same problem

melonholy commented on src/containers/auth/resetPassword/index.js:
```diff
@@ -0,0 +1,186 @@
+  }
+
+  showCurrentMessage = (
+    keys = [],
```
Comments above

melonholy commented on src/containers/auth/resetPassword/index.js:
```diff
@@ -0,0 +1,186 @@
+  };
+
+  componentDidMount() {
+    const params = new URLSearchParams(location.search);
```
use `query-string` library

OlegPanasyuk replied:
Done

melonholy commented on src/containers/auth/resetPassword/styles.module.scss:
```diff
@@ -0,0 +1,60 @@
+  }
+
+  .row {
+    margin: 15px 0 0 0;
```
use `margin-top`

melonholy commented on src/containers/auth/resetPassword/styles.module.scss:
```diff
@@ -0,0 +1,60 @@
+  }
+
+  .row_top {
+    margin: 25px 0 0 0;
```
comment above

melonholy commented on src/containers/auth/styles.module.scss:
```diff
@@ -64,6 +66,23 @@
+    color: $white;
+    opacity: 0.6;
+    cursor: pointer;
+    margin: 0 0 25px 0;
```
use `margin-bottom`

OlegPanasyuk replied:
Done

OlegPanasyuk committed:
Fix Vlad's notes

slavasgod commented on src/components/spinner/index.js:
```diff
@@ -0,0 +1,12 @@
+        return (
+            <CircleSpinner size={40} color="#fff" loading={this.props.loading}/>
+        );
+    }
```
Use prettier.

OlegPanasyuk replied:
Done

slavasgod commented on src/containers/auth/forgotPassword/index.js:
```diff
@@ -0,0 +1,176 @@
+import styles from './styles.module.scss';
+import { forgotPasswordRequest } from '../../../services/auth';
+import validateAuth from '../../../validation/auth';
+import ShowMessage from '../showMessage';
```
Fix import order.

slavasgod commented on src/containers/auth/resetPassword/index.js:
```diff
@@ -0,0 +1,206 @@
+import { passwordValidation } from '../../../validation/auth';
+import ShowMessage from '../showMessage';
+import Spinner from '../../../components/spinner';
+import Form from '../../../components/form';
```
Fix import order

slavasgod commented on src/containers/auth/styles.module.scss:
```diff
@@ -64,6 +66,23 @@
+    margin-bottom: 25px;
+    transition-property: all;
+    transition-duration: .1s;
+
```
Just in case, you can combine these transition properties in one:
`transition: all .1s;`

OlegPanasyuk replied:
Done

slavasgod commented on src/services/auth.js:
```diff
@@ -44,5 +44,59 @@ export const googleLoginRequest = async data => {
+    return {
+      status: 400,
+      data: 'Bad email',
+    };
```
Make normal answer on backend side. Then you can easily catch it in components, where you call this function

OlegPanasyuk replied:
Done

slavasgod replied:
!

slavasgod commented on src/services/auth.js:
```diff
@@ -44,5 +44,59 @@ export const googleLoginRequest = async data => {
+    return {
+      status: 400,
+      data: 'Something is wrong',
+    };
```
Make normal answer

OlegPanasyuk replied:
Done

OlegPanasyuk committed:
Merge dev into task branch

OlegPanasyuk committed:
Fix Slava's Notes
